plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+    {
+        DeleteTableRequest deleteNewTableRequest = new DeleteTableRequest()
+                .withDatabaseName(schema)
+                .withName(table);
+        catalogId.ifPresent(deleteNewTableRequest::setCatalogId);
+        stats.getDropTable().call(() -> glueClient.deleteTable(deleteNewTableRequest));
+    }
+
+    @Override
+    public void updateTableComment(ConnectorSession session, SchemaTableName schemaTableName, Optional<String> comment)
+    {
+        Table icebergTable = loadTable(session, schemaTableName);
+        if (comment.isEmpty()) {
+            icebergTable.updateProperties().remove(TABLE_COMMENT).commit();
+        }
+        else {
+            icebergTable.updateProperties().set(TABLE_COMMENT, comment.get()).commit();
+        }
+    }
+
+    @Override
+    public String defaultTableLocation(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        GetDatabaseRequest getDatabaseRequest = new GetDatabaseRequest()
+                .withName(schemaTableName.getSchemaName());
+        catalogId.ifPresent(getDatabaseRequest::setCatalogId);
+        String databaseLocation = stats.getGetDatabase().call(() ->
+                glueClient.getDatabase(getDatabaseRequest)
+                        .getDatabase()
+                        .getLocationUri());
+
+        String tableName = schemaTableName.getTableName();
+        if (isUniqueTableLocation) {
+            tableName = tableName + "-" + randomUUID().toString().replace("-", "");
+        }
+
+        Path location;
+        if (databaseLocation == null) {
+            if (defaultSchemaLocation.isEmpty()) {
+                throw new TrinoException(
+                        HIVE_DATABASE_LOCATION_ERROR,
+                        format("Schema '%s' location cannot be determined. " +
+                                        "Either set the 'location' property when creating the schema, or set the 'hive.metastore.glue.default-warehouse-dir' " +
+                                        "config option.",
+                                schemaTableName.getSchemaName()));
+            }
+            String schemaDirectoryName = schemaTableName.getSchemaName() + ".db";
+            location = new Path(new Path(defaultSchemaLocation.get(), schemaDirectoryName), tableName);
+        }
+        else {
+            location = new Path(databaseLocation, tableName);
+        }
+
+        return location.toString();
+    }
+
+    @Override
+    public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTableName, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setTablePrincipal is not supported for Iceberg Glue catalogs");
+    }
+
+    @Override
+    public void createView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorViewDefinition definition, boolean replace)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "createView is not supported for Iceberg Glue catalogs");
```

Comment:
Iceberg connector support Trino views.
Please implement this (might be as follow-up)